entry.go

```diff
@@ -82,7 +82,10 @@ func (entry *Entry) WithFields(fields Fields) *Entry {
 	for k, v := range fields {
 		data[k] = v
 	}
-	return &Entry{Logger: entry.Logger, Data: data}
+	// shallow copy
+	newentry := *entry
+	newentry.Data = data
+	return &newentry
```

Comment:
The unused fields are zero initialized anyway, so this shouldn't result in any significant performance penalties. Benchmark it maybe?